### PR TITLE
Add tls for frontgate and pilot

### PR DESCRIPTION
**Pull request by Hevienz**

### CODE REVIEW

File: deploy/kubernetes/openpitrix/metadata/openpitrix-pilot.yaml [hunk 1]
```diff
@@ -17,6 +17,10 @@ spec:
     protocol: TCP
     port: 9110
     targetPort: 9110
+  - name: for-frontgate
```
Comment (chilianyi) [inline deploy/kubernetes/openpitrix/metadata/openpitrix-pilot.yaml]: `for-frontgate` don't feel good.

How about openpitrix-pilot-tls

File: deploy/kubernetes/openpitrix/metadata/openpitrix-pilot.yaml [hunk 2]
```diff
@@ -53,10 +57,12 @@ spec:
       - name: openpitrix-pilot
         image: ${IMAGE}
         imagePullPolicy: ${IMAGE_PULL_POLICY}
-        command: ['pilot', '-config=/opt/openpitrix/conf/pilot-config.json', 'serve']
+        command: ['pilot', '-config=/opt/openpitrix/conf/pilot-config.json', '--openpitrix-ca-crt-file=/opt/openpitrix/conf/openpitrix-ca.crt', '--pilot-server-crt-file=/opt/openpitrix/conf/pilot-server.crt', '--pilot-server-key-file=/opt/openpitrix/conf/pilot-server.key', '--pilot-client-crt-file=/opt/openpitrix/conf/pilot-client.crt', '--pilot-client-key-file=/opt/openpitrix/conf/pilot-client.key', 'serve']
         ports:
         - containerPort: 9110
           name: pilot
+        - containerPort: 9114
+          name: for-frontgate
```
Comment (chilianyi) [inline deploy/kubernetes/openpitrix/metadata/openpitrix-pilot.yaml]: pilot-tls

File: deploy/kubernetes/scripts/deploy-k8s.sh [hunk 3]
```diff
@@ -90,7 +90,7 @@ cd ../..
 
 kubectl create namespace ${NAMESPACE}
 kubectl create secret generic mysql-pass --from-file=./kubernetes/password.txt -n ${NAMESPACE}
-
+./kubernetes/scripts/generate-certs.sh -n ${NAMESPACE}
```
Comment (chilianyi) [inline deploy/kubernetes/scripts/deploy-k8s.sh]: Move this line into `if [ "${METADATA}" == "1" ];then`

File: deploy/kubernetes/scripts/generate-certs.sh [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+do
+  case "${option}"
+  in
+    n) NAMESPACE=${OPTARG};;
+  esac
+done
+
+cd $(dirname $0)
+cd ../../..
+make generate-certs
```
Comment (chilianyi) [inline deploy/kubernetes/scripts/generate-certs.sh]: Our deploy package have not source code.

File: deploy/kubernetes/openpitrix/metadata/openpitrix-pilot.yaml (hunk 1 above)
Comment (Hevienz) [reply deploy/kubernetes/openpitrix/metadata/openpitrix-pilot.yaml]: ok

File: deploy/kubernetes/openpitrix/metadata/openpitrix-pilot.yaml [hunk 5]
```diff
@@ -90,5 +108,6 @@ data:
         "id": "openpitrix-pilot-001",
         "host": "localhost",
         "listen_port": 9110,
+        "for_frontgate_listen_port": 9114,
```
Comment (chilianyi) [inline deploy/kubernetes/openpitrix/metadata/openpitrix-pilot.yaml]: Rename to tls_listen_port is better.

File: metadata/cmd/pilot/tls-config/Makefile [hunk 6]
```diff
@@ -37,7 +37,7 @@ pilot-client.key:
 
 pilot-client.crt: pilot-client.key openpitrix-ca.key openpitrix-ca.crt
 	openssl req -new \
-		-subj "/C=GB/L=China/O=pilot-client/CN=pilot-client.openpitrix.io" \
```
Comment (chai2010) [inline metadata/cmd/pilot/tls-config/Makefile]: the "pilot.openpitrix.io" is pilot server name, client common name is different.

**Pull request merged**